engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/stores/ExecutionStore.scala

```diff
@@ -130,6 +130,10 @@ sealed abstract class ExecutionStore private[stores](statusStore: Map[JobKey, Ex
   // View of the statusStore more suited for lookup based on status
   lazy val store: Map[ExecutionStatus, List[JobKey]] = statusStore.groupBy(_._2).mapValues(_.keys.toList)
 
+  def keyForNode(node: GraphNode): Option[JobKey] = {
```

Comment:
ToL if it doesn't break an abstraction layer, maybe this fits in `EnhancedGraphNode` above?

Reply:
Yes exactly. I wanted to move that other big chunk of logic over here as well and had a similar problem.